packages/dev/addons/src/navigation/plugin/RecastJSCrowd.ts

```diff
@@ -0,0 +1,439 @@
+import type { Crowd } from "@recast-navigation/core";
+
+import type { Nullable } from "core/types";
+import type { IVector3Like } from "core/Maths/math.like";
+import type { TransformNode } from "core/Meshes/transformNode";
+import type { ICrowd } from "core/Navigation/INavigationEngine";
+import { Vector3 } from "core/Maths/math.vector";
+import { Epsilon } from "core/Maths/math.constants";
+import type { Observer } from "core/Misc/observable";
+import { Observable } from "core/Misc/observable";
+import type { Scene } from "core/scene";
+
+import type { RecastNavigationJSPluginV2 } from "./RecastNavigationJSPlugin";
+import type { AbstractEngine } from "core/Engines/abstractEngine";
+import { ToCrowdAgentParams } from "../common";
+import type { IAgentParametersV2 } from "../types";
+import { BjsRecast } from "../factory/common";
+
+/**
+ * Recast Detour crowd implementation
+ * This class provides methods to manage a crowd of agents, allowing them to navigate a navigation mesh.
+ * It supports adding agents, updating their parameters, moving them to destinations, and checking their states.
+ * The crowd is updated in the scene's animation loop, and it notifies observers when agents reach their destinations.
+ */
+export class RecastJSCrowd implements ICrowd {
+    /**
+     * Recast plugin
+     */
+    public navigationPlugin: RecastNavigationJSPluginV2;
+    /**
+     * Link to the detour crowd
+     */
+    public recastCrowd: Crowd;
+    /**
+     * One transform per agent
+     */
+    public transforms: TransformNode[] = new Array<TransformNode>();
+    /**
+     * All agents created
+     */
+    public agents: number[] = new Array<number>();
+    /**
+     * agents reach radius
+     */
+    public reachRadii: number[] = new Array<number>();
```

Comment:
Maybe make any of these private that are not intended to be used outside the class. If they are intended to be used outside, make them readonly if possible, otherwise private with a public getter.

Reply:
Umm I'm not following. This is the concrete implementation of an interface. We aren't trying to have back compat between two concrete implementations, just back compat via the interface, right? These members are not part of the interface, so it seems like it should not be a problem to handle them differently than the other implementation, or am I still missing something?

Reply:
Backwards compatiblity must follow not just the interface but the current implementation as well. Just imagine someone is R/W accessing those `public` variables directly in his code. Thinking...  :-D Ok, since these are all refernces a getter should not cause any trouble here... I am sorry, I didn't really think this trough in the previous round... I'm going to change these to private/getter.